### ISSUE TITLE
Uninitialized readonly property is not reported

### ISSUE THREAD

**Issue by dobrakmato** [Microsoft/TypeScript#18667]
**TypeScript Version:**  2.5.2

**Code**

```ts
class Connection {
    public query(sql) {
        return null;
    }
}

class Database {
    private readonly property: string = 'This is initialized';
    private readonly thisToo: number;
    private readonly connection: Connection;

    constructor() {
        this.thisToo = 42;
    }

    public importantMethod() {
        return this.connection.query('SELECT * FROM users');
    }
}
```

**Expected behavior:**
I expect the property `connection` to be reported as **"Not initialized read-only property"** error because it is 
1. not initialized in constructor
2. also marked as `readonly` so **it can't be initialized** anywhere else.

**Actual behavior:**
Compiler generates broken / not working code.

```js
var Database = /** @class */ (function () {
    function Database() {
        this.property = 'This is initialized';
        this.thisToo = 42;
    }
    Database.prototype.importantMethod = function () {
        return this.connection.query('SELECT * FROM users');
    };
    return Database;
}());
```

**Comment by mhegazy** [comment]
This should be done along with https://github.com/Microsoft/TypeScript/issues/8476. basically same analysis required in both cases.

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
